core-java/src/test/java/org/baeldung/java/JavaStringConversionUnitTest.java

```diff
@@ -0,0 +1,193 @@
+        Byte wrapperByte = Byte.valueOf("1");
+
+        assertThat(wrapperByte == 1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveShort_thenCorrect() {
+        short primitiveShort = Short.parseShort("1");
+
+        assertThat(primitiveShort == 1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToWrapperShort_thenCorrect() {
+        Short wrapperShort = Short.valueOf("1");
+
+        assertThat(wrapperShort == 1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveInt_thenCorrect() {
+        int primitiveInt = Integer.parseInt("1");
+
+        assertThat(primitiveInt == 1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToWrapperInteger_thenCorrect() {
+        Integer wrapperInt = Integer.valueOf("1");
+
+        assertThat(wrapperInt == 1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveLong_thenCorrect() {
+        long primitiveLong = Long.parseLong("1");
+
+        assertThat(primitiveLong == 1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToWrapperLong_thenCorrect() {
+        Long wrapperLong = Long.valueOf("1");
+
+        assertThat(wrapperLong == 1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveFloat_thenCorrect() {
+        float primitiveFloat = Float.parseFloat("1.1");
+
+        assertThat(primitiveFloat == 1.1F).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToWrapperFloat_thenCorrect() {
+        Float wrapperFloat = Float.valueOf("1.1");
+
+        assertThat(wrapperFloat == 1.1F).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveDouble_thenCorrect() {
+        double primitiveDouble = Double.parseDouble("1.1");
+
+        assertThat(primitiveDouble == 1.1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToWrapperDouble_thenCorrect() {
+        Double wrapperDouble = Double.valueOf("1.1");
+
+        assertThat(wrapperDouble == 1.1).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveChar_thenCorrect() {
+        String primitiveChar = "C";
+        char h = primitiveChar.charAt(0);
+
+        assertThat(h == 'C').isTrue();
+    }
+
+
+    @Test
+    public void whenConvertingStringToPrimitiveCharArray_thenCorrect() {
+        String primitiveChars = "Hello world";
+        char[] chars = primitiveChars.toCharArray();
+
+        assertThat(Arrays.equals(chars,new char[]{'H','e', 'l', 'l', 'o', ' ', 'w', 'o', 'r', 'l', 'd'})).isTrue();
+    }
+
+    @Test
+    public void whenConvertingStringToPrimitiveByteArray_default_thenCorrect() {
+        String primitiveBytes = "Hello world";
+        byte[] bytes = primitiveBytes.getBytes();
+
+        System.out.println(Arrays.toString(bytes));
+    }
+
+    @Test
+    public void stringToPrimitiveByteArray_CharsetName_thenCorrect() throws UnsupportedEncodingException {
```

Comment:
Similarly, we can say - **givenString_whenConvertingToByteArrayWithCharsetName_thenCorrect**